src/main/application/services/tcdata/GetTCDataUseCase.js

```diff
@@ -0,0 +1,89 @@
+import {inject} from '../../../core/ioc/ioc'
+import {TCData} from '../../../domain/tcdata/TCData'
+import {ConsentRepository} from '../../../domain/consent/ConsentRepository'
+import {ConsentDecoderService} from '../../../domain/consent/ConsentDecoderService'
+import {StatusRepository} from '../../../domain/status/StatusRepository'
+export class GetTCDataUseCase {
+  /**
+   *
+   * @param {Object} param
+   * @param {StatusRepository} param.statusRepository
+   */
+  constructor({
+    statusRepository = inject(StatusRepository),
+    consentRepository = inject(ConsentRepository),
+    consentDecoderService = inject(ConsentDecoderService)
+  } = {}) {
+    this._statusRepository = statusRepository
+    this._consentRepository = consentRepository
+    this._consentDecoderService = consentDecoderService
+  }
+
+  /**
+   *
+   * @param {Object} param
+   * @param {Array<Number>} param.vendorIds
+   */
+  execute({vendorIds} = {}) {
+    if (
+      Array.isArray(vendorIds) &&
+      vendorIds.some(
+        vendorId =>
+          typeof vendorId !== 'number' ||
+          Number.isInteger(vendorId) ||
+          vendorId < 1
+      )
+    ) {
+      throw Error(
+        'vendorIds parameter of getTCData has invalid data. It must be an array of positive integers, or undefined'
+      )
+    } else if (vendorIds !== undefined && !Array.isArray(vendorIds)) {
+      throw Error(
+        'vendorIds, if defined, needs to be an array of positive integers'
+      )
```

Comment:
Do we need two different errors? Wouldn't it be the same error?

Reply:
@maryammdot I tried with one condition and It's not easy to understand. I split in too only for readability